enhancements/sdk-supports-operands-deletion.md

```diff
@@ -0,0 +1,135 @@
+
+To enable this, it would be ideal to have a finalizer "no-op" handler, i.e. some code that does not clean up a CR but has a `TODO(user)` comment, scaffolded by default in the kubebuilder project. The idea is to always scaffold out a finalizer, a delete event handler and a Reconciler that checks for Finalizers and has a no-op in cases where it's not required to use a Finalizer.
+
+The Reconciler checks for presence of a Finalizer in a CR object. If the CR object is marked for deletion, indicated by `metadata.deletionTimestamp` being set, the Reconciler runs the finilizer logic and removes the finalizer. If the finalization logic fails for some reason, then the Reconciler keeps the finalizer to retry during the next reconcilation. Typically, the finalizer logic would entail necessary cleanup steps that the operator needs to perform before the CR can be deleted. See [finalizer example][finalizer-example] for more details.
+
+#### Configuration of the cleanup API in SDK
+
+A [ClusterServiceVersion (CSV)][csv] represents a particular version of a running operator on a cluster and is an actual installation of the operator. The CSV will support the `spec.cleanup` field once the [operand cleanup proposal][operand-cleanup-proposal] is implemented
+
+To enable configuration of operand deletion on the CSV, the `operator-sdk generate kustomize manifests` command will generate CSVs with the `spec.cleanup` field set to `false` by default. The user can then explicitly set this field to true to opt-in to automatic deletion of CRs using OLM.
+
+#### Existing cleanup command in SDK
+
+The existing `operator-sdk cleanup` command currently deletes the CRDs and all other resources owned by the operator, without an option to disable this behavior. Ideally, the cleanup command should have certain flags that prevent deletion of CRDs, for feature parity with the [kubectl operator plugin](https://github.com/operator-framework/kubectl-operator/).
+
+Existing cleanup command in OSDK:
+
+```
+operator-sdk cleanup <operatorPackageName>
+```
+
+Proposed cleanup command with flags:
+
+```
+operator-sdk cleanup <operatorPackageName> --delete-crds --delete-all
+```
+
+`--delete-crds` boolean flag should be added to the `cleanup` command, and defaulted to `true` to preserve current behavior. When `--delete-crds=false`, the command will not clean up CRDs.
+
+Additionally, `--delete-all` flag should also be added on the `cleanup` command that will delete all owned CRDs, CRs and operator groups.
+
+These flags can be set accordingly to prevent deletion of CRDs unless explicitly asked for by the user.
```

Comment:
With only flag `--delete-crds`, it won't behave too differently, as we will default to removing everything to preserve current behavior if the flag is not set. But still, it would be a breaking change.